cni-plugin/pkg/plugin/plugin.go

```diff
@@ -104,6 +110,29 @@ func testConnection() error {
 	return nil
 }
 
+func isEndpointReady(readyEndpoint string) (bool, error) {
+	if readyEndpoint == "" {
+		readyEndpoint = readinessEndpoint
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), pollInterval)
```

Comment:
could we have a `timeout` arg passed to `isEndpointReady()` and use it here?

Reply:
Done